apps/api/src/function/src/engine.ts

```diff
@@ -241,18 +248,26 @@ export class FunctionEngine implements OnModuleInit, OnModuleDestroy {
         if (!isCorrectDepth || !isTrackedFile) return;
 
         const dirName = parts[0];
+
+        let content: string;
+        if (type == "delete") {
+          content = null;
+        } else {
+          const b = await fs.promises.readFile(path);
+          content = b.toString();
+        }
+
         const fn = await CRUD.findByName(this.fs, dirName, {
           resolveEnvRelations: EnvRelation.NotResolved
         });
         if (!fn) return;
 
-        const content = await fs.promises.readFile(path).then(b => b.toString());
-
-        observer.next({...fn, content});
+        observer.next({...fn, content, type});
```

Comment:
It's better to use promise all for these two promises instead of waiting one by one